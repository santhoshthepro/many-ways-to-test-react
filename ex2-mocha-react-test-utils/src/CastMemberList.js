var React = require('react');
var CastMember = require('./CastMember');

var CastMemberList = React.createClass({
  render: function() {
    const castMembers = this.props.members.map((member) => {
      return <CastMember key={member.id} member={member} />
    });

    return (
      <div>
        {castMembers}
      </div>
    );
  }
});

module.exports = CastMemberList;
