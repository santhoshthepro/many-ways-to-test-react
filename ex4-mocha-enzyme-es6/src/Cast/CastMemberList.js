import React, {PropTypes} from 'react';
import CastMember from './CastMember';

const CastMemberList = ({ members }) => {
  const castMembers = members.map((member) => {
    return <CastMember key={member.id} member={member} />
  });

  return (
    <div className="member-list">
      {castMembers}
    </div>
  );
};

CastMemberList.propTypes = {
  members: PropTypes.array.isRequired
};

export default CastMemberList;
