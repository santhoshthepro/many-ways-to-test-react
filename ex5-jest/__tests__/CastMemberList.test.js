import React from 'react';
import renderer from 'react-test-renderer';
import CastMemberList from '../src/Cast/CastMemberList';

describe('CastMemberList', function() {
  let props;

  beforeEach(function() {
    props = {
      members: [
        {
          id: 3,
          name: 'Yul Brynner',
          imageUrl: 'yul-brynner.jpg',
          thumbnailUrl: 'yul-brynne-small.jpg',
          bio: 'Exotic leading man of American films, famed as much for his completely bald head as for his performances, Yul Brynner masked much of his life in mystery and outright lies designed to tease people he considered gullible.'
        },
        {
          id: 6,
          name: 'Deborah Kerr',
          imageUrl: 'deborah-kerr.jpg',
          thumbnailUrl: 'deborah-kerr-small.jpg',
          bio: 'Deborah Kerr was born on 30 September 1921 in Helensburgh, Scotland, the daughter of Captain Arthur Kerr-Trimmer.'
        }
      ]
    };
  });

  test('CastMemberList rendered static HTML', () => {
    const component = renderer.create(
      <CastMemberList {...props} />
    );

    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
