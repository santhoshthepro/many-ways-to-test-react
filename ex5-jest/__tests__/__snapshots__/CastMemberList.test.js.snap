exports[`CastMemberList CastMemberList rendered static HTML 1`] = `
<div
  className="member-list">
  <div
    className="media">
    <div
      className="media-left">
      <a
        href="#">
        <img
          alt="Yul Brynner"
          className="media-object"
          src="/images/yul-brynne-small.jpg" />
      </a>
    </div>
    <div
      className="media-body">
      <h2
        className="media-heading">
        Yul Brynner
      </h2>
      Exotic leading man of American films, famed as much for his completely bald head as for his performances, Yul Brynner masked much of his life in mystery and outright lies designed to tease people he considered gullible.
    </div>
  </div>
  <div
    className="media">
    <div
      className="media-left">
      <a
        href="#">
        <img
          alt="Deborah Kerr"
          className="media-object"
          src="/images/deborah-kerr-small.jpg" />
      </a>
    </div>
    <div
      className="media-body">
      <h2
        className="media-heading">
        Deborah Kerr
      </h2>
      Deborah Kerr was born on 30 September 1921 in Helensburgh, Scotland, the daughter of Captain Arthur Kerr-Trimmer.
    </div>
  </div>
</div>
`;
